modules/l4clock/matcher.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2024 VNXME
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4clock
+
+import (
+	"strings"
+	"time"
+	_ "time/tzdata"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+
+	"github.com/mholt/caddy-l4/layer4"
+)
+
+func init() {
+	caddy.RegisterModule(&MatchClock{})
+}
+
+// MatchClock is able to match any connections using the time when they are wrapped/matched.
+type MatchClock struct {
+	After    string `json:"after,omitempty"`
+	Before   string `json:"before,omitempty"`
+	Timezone string `json:"timezone,omitempty"`
```

Comment:
The expected format(s) should be documented here, maybe an example as well. `HH:MM:SS` (24-hour format).

Also, I find it easier to specify UTC offsets for timezones, or 3-letter tz abbreviations.. like "-6:00" or "MST" or something like that. That's just my preference and I have no idea the implications of allowing that, but... what are your thoughts?

Reply:
I've provided comments for the matcher fileds. Besides, I've added custom offset support (e.g -06:00 is a valid timezone value now). 3-letter timezone abbreviations are ambiguous and not included in the standard time library.